CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -257,7 +257,10 @@ public extension WorkspaceClient {
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
                 let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let filextension = fileUrl.pathExtension
+                let fileName = filextension.isEmpty ? previousName : previousName.replacingOccurrences(of: ".\(filextension)", with: "")
+                fileUrl = fileUrl.deletingLastPathComponent()
+                fileurl.appendPathComponent("\(fileName) copy.\(filextension)")
```

Comment:
`fileurl` should be `fileUrl`

Please make sure your code compiles before pushing it.

Reply:
Alright; I think I might have accidentally while I was trying to commit, since it compiled and ran fine whenever I tested it. My bad